drivers/adc/ad7124/ad7124.c

```diff
@@ -68,6 +79,7 @@ int32_t ad7124_no_check_read_register(struct ad7124_dev *dev,
 	uint8_t i = 0;
 	uint8_t check8 = 0;
 	uint8_t msg_buf[8] = {0, 0, 0, 0, 0, 0, 0, 0};
+	int32_t add_status_length = 0;
```

Comment:
this is not just cleanup, you actually add functionality here. I would do this in a separate commit. Also uint8_t will be enough and you can pair it with check8.

uint8_t check8 = 0, add_status_length = 0;

Reply:
will review this pull request so it covers just the adding support appending 4th byte of STATUS to DATA